From: =?utf-8?q?Christian_G=C3=B6ttsche?= <cgzones@googlemail.com>
Date: Fri, 14 Aug 2020 11:41:00 +0200
Subject: test-mech.c: fix 32-bit issues
Forwarded: https://github.com/dovecot/core/pull/134
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

Use size_t and %zu for sizes as on 32bit architectures sizes are not of type unsigned long.

test-mech.c: In function ‘test_mechs’:
test-mech.c:326:61: warning: format ‘%lu’ expects argument of type ‘long unsigned int’, but argument 4 has type ‘unsigned int’ [-Wformat=]
  326 |   const char *testname = t_strdup_printf("auth mech %s %d/%lu",
      |                                                           ~~^
      |                                                             |
      |                                                             long unsigned int
      |                                                           %u
test-mech.c:338:12: warning: passing argument 2 of ‘test_mech_construct_apop_challenge’ from incompatible pointer type [-Wincompatible-pointer-types]
  338 |            &test_case->len);
      |            ^~~~~~~~~~~~~~~
      |            |
      |            size_t * {aka unsigned int *}
test-mech.c:195:77: note: expected ‘long unsigned int *’ but argument is of type ‘size_t *’ {aka ‘unsigned int *’}
  195 | test_mech_construct_apop_challenge(unsigned int connect_uid, unsigned long *len_r)
      |                                                              ~~~~~~~~~~~~~~~^~~~~
---
 src/auth/test-mech.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/auth/test-mech.c b/src/auth/test-mech.c
index cf05370..db9f85c 100644
--- a/src/auth/test-mech.c
+++ b/src/auth/test-mech.c
@@ -192,11 +192,11 @@ static void test_mech_handle_challenge(struct auth_request *request,
 }
 
 static inline const unsigned char *
-test_mech_construct_apop_challenge(unsigned int connect_uid, unsigned long *len_r)
+test_mech_construct_apop_challenge(unsigned int connect_uid, size_t *len_r)
 {
 	string_t *apop_challenge = t_str_new(128);
 
-	str_printfa(apop_challenge,"<%lx.%u.%"PRIdTIME_T"", (unsigned long) getpid(),
+	str_printfa(apop_challenge,"<%lx.%u.%"PRIxTIME_T"", (unsigned long) getpid(),
 		    connect_uid, process_start_time+10);
 	str_append_data(apop_challenge, "\0testuser\0responseoflen16-", 26);
 	*len_r = apop_challenge->used;
@@ -323,7 +323,7 @@ static void test_mechs(void)
 		struct test_case *test_case = &tests[running_test];
 		const struct mech_module *mech = test_case->mech;
 		struct auth_request *request;
-		const char *testname = t_strdup_printf("auth mech %s %d/%lu",
+		const char *testname = t_strdup_printf("auth mech %s %d/%zu",
 						       mech->mech_name,
 						       running_test+1,
 						       N_ELEMENTS(tests));
